mlflow/data/spark_dataset.py

```diff
@@ -119,7 +119,7 @@ def profile(self) -> Optional[Any]:
             drdd = py_rdd.mapPartitions(lambda it: [float(sum(1 for i in it))])
             jrdd = drdd.mapPartitions(lambda it: [float(sum(it))])._to_java_object_rdd()
             jdrdd = drdd.ctx._jvm.JavaDoubleRDD.fromRDD(jrdd.rdd())
-            timeout_millis = 2000
+            timeout_millis = 5000
```

Comment:
5 seconds hopefully shouldn't be too long. we can decrease this later if necessary